docs/building-apps/example-application-tutorial/querying-data.mdx

```diff
@@ -0,0 +1,335 @@
+ * @async
+ * @function findStrictSendPaths
+ * @param {Object} opts Options object
+ * @param {string} opts.sourceAsset Stellar asset which will be sent from the source account
+ * @param {string|number} opts.sourceAmount Amount of the Stellar asset that should be debited from the srouce account
+ * @param {string} opts.destinationPublicKey Public Stellar address that will receive the destination asset
+ * @returns {Promise<PaymentPathRecord[]>} Array of payment paths that can be selected for the transaction
+ * @throws Will throw an error if there are no available payment paths.
+ */
+export async function findStrictSendPaths({
+  sourceAsset,
+  sourceAmount,
+  destinationPublicKey,
+}) {
+  let asset =
+    sourceAsset === "native"
+      ? Asset.native()
+      : new Asset(sourceAsset.split(":")[0], sourceAsset.split(":")[1]);
+  let response = await server
+    .strictSendPaths(asset, sourceAmount.toString(), destinationPublicKey)
+    .call();
+  if (response.records.length > 0) {
+    return response.records;
+  } else {
+    throw error(400, { message: "no strict send paths available" });
+  }
+}
+```
+
+**Source:** https://github.com/stellar/basic-payment-app/blob/main/src/lib/stellar/horizonQueries.js
+
+## findStrictReceivePaths
+
+Find the available strict receive paths between a source account and receiving asset/amount.
+
+```js title="/src/lib/stellar/horizonQueries.js"
+/**
+ * Fetches available paths on the Stellar network between the source account, and the asset to be received by the destination.
+ * @async
+ * @function findStrictReceivePaths
+ * @param {Object} opts Options object
+ * @param {string} opts.sourcePublicKey Public Stellar address that will be the source of the payment operation
+ * @param {string} opts.destinationAsset Stellar asset which should be received in the destination account
+ * @param {string|number} opts.destinationAmount Amount of the Stellar asset that should be credited to the destination account
+ * @returns {Promise<PaymentPathRecord[]>} Array of payment paths that can be selected for the transaction
+ * @throws Will throw an error if there are no available payment paths.
+ */
+export async function findStrictReceivePaths({
+  sourcePublicKey,
+  destinationAsset,
+  destinationAmount,
+}) {
+  let asset =
+    destinationAsset === "native"
+      ? Asset.native()
+      : new Asset(
+          destinationAsset.split(":")[0],
+          destinationAsset.split(":")[1],
+        );
+  let response = await server
+    .strictReceivePaths(sourcePublicKey, asset, destinationAmount.toString())
+    .call();
+  if (response.records.length > 0) {
+    return response.records;
+  } else {
+    throw error(400, { message: "no strict receive paths available" });
+  }
+}
+```
+
+**Source:** https://github.com/stellar/basic-payment-app/blob/main/src/lib/stellar/horizonQueries.js
+
+## Query Stellar Expert
```

Comment:
Does this copy make sense?

Reply:
Not really, it didn't lol. I'm pushing some better writing now.